File: src/syntheticExamples/scopeFunctions/CFR/ScopeFunctionsDecompiled/ScopeFunctionsKt.java
package syntheticExamples.scopeFunctions.CFR.ScopeFunctionsDecompiled;

import kotlin.Metadata;

@Metadata(mv={2, 0, 0}, k=2, xi=82, d1={"\u0000\b\n\u0000\n\u0002\u0010\u0002\n\u0000\u001a\u0006\u0010\u0000\u001a\u00020\u0001\u00a8\u0006\u0002"}, d2={"main", "", "KotlinDecompiler"})
public final class ScopeFunctionsKt {
    public static final void main() {
        Person updatedPerson;
        Person person;
        Person person2;
        Person $this$main_u24lambda_u240 = person2 = (person = new Person("Unknown", 0, null));
        boolean bl = false;
        $this$main_u24lambda_u240.setName("Alice");
        $this$main_u24lambda_u240.setAge(30);
        Person it = updatedPerson = person2;
        boolean bl2 = false;
        System.out.println((Object)("Person's name is " + it.getName()));
        int nameLength = it.getName().length();
        System.out.println((Object)("Name length: " + nameLength));
        Person $this$main_u24lambda_u242 = updatedPerson;
        boolean bl3 = false;
        System.out.println((Object)("Checking if " + $this$main_u24lambda_u242.getName() + " is adult."));
        boolean isAdult = $this$main_u24lambda_u242.getAge() >= 18;
        System.out.println((Object)("Is adult: " + isAdult));
        Person it2 = it = updatedPerson;
        boolean bl4 = false;
        System.out.println((Object)("Sending email to " + it2.getEmail()));
        Person $this$main_u24lambda_u244 = it;
        boolean bl5 = false;
        $this$main_u24lambda_u244.setEmail("alice@example.com");
        Person $this$main_u24lambda_u245 = updatedPerson;
        boolean bl6 = false;
        String contactInfo = $this$main_u24lambda_u245.getName() + " can be contacted at " + $this$main_u24lambda_u245.getEmail();
        System.out.println((Object)contactInfo);
    }

    public static /* synthetic */ void main(String[] args) {
        ScopeFunctionsKt.main();
    }
}

File: src/syntheticExamples/scopeFunctions/CFR/ScopeFunctionsJ2K/ScopeFunctionsKt.kt
package syntheticExamples.scopeFunctions.CFR.ScopeFunctionsJ2K

object ScopeFunctionsKt {
    fun main() {
        val updatedPerson: Person
        var person: Person?
        val person2 =
            (Person("Unknown", 0, null).also { person = it })
        val `$this$main_u24lambda_u240` = person2
        val bl = false
        `$this$main_u24lambda_u240`.name = "Alice"
        `$this$main_u24lambda_u240`.age = 30
        updatedPerson = person2
        var it = updatedPerson
        val bl2 = false
        println(("Person's name is " + it.name) as Any)
        val nameLength = it.name.length
        println(("Name length: $nameLength") as Any)
        val `$this$main_u24lambda_u242` = updatedPerson
        val bl3 = false
        println(("Checking if " + `$this$main_u24lambda_u242`.name + " is adult.") as Any)
        val isAdult = `$this$main_u24lambda_u242`.age >= 18
        println(("Is adult: $isAdult") as Any)
        it = updatedPerson
        val it2 = it
        val bl4 = false
        println(("Sending email to " + it2.email) as Any)
        val `$this$main_u24lambda_u244` = it
        val bl5 = false
        `$this$main_u24lambda_u244`.email = "alice@example.com"
        val `$this$main_u24lambda_u245` = updatedPerson
        val bl6 = false
        val contactInfo = `$this$main_u24lambda_u245`.name + " can be contacted at " + `$this$main_u24lambda_u245`.email
        println(contactInfo as Any)
    }

    @JvmStatic
    fun main(args: Array<String>) {
        main()
    }
}